### DISCUSSION TITLE
Triger workflows with http asynchronously

### DISCUSSION THREAD

**Discussion by badihi** [elsa-workflows/elsa-core#988] [Q&A]
I'm looking for a way to invoke a long-running workflow with http. So the connection closes immediately after calling the http endpoint (and maybe returns an id of the started instance), and the execution continues in background,

Is there anyway to do this in Elsa 2?

**Comment by sfmskywalker** [comment]
Yep, this feature was just merged today 🥳 .

To invoke a long-running HTTP workflow, simply supply the `x-dispatch` query string parameter or as a request header.
Sample:

```
/my-http-workflow?x-dispatch
```

**Reply by badihi** [reply]
Thanks very much for your attention. I was trying to test this feature on rc95, but now I face this exception on startup (on aspnet):

```
System.AggregateException: 'Some services are not able to be constructed (Error while validating the service descriptor 'ServiceType: Elsa.Activities.Signaling.Services.ISignaler Lifetime: Scoped ImplementationType: Elsa.Activities.Signaling.Services.Signaler': Unable to resolve service for type 'Elsa.Services.ITokenService' while attempting to activate 'Elsa.Activities.Signaling.Services.Signaler'.) (Error while validating the service descriptor 'ServiceType: Elsa.Activities.Signaling.SendSignal Lifetime: Transient ImplementationType: Elsa.Activities.Signaling.SendSignal': Unable to resolve service for type 'Elsa.Services.ITokenService' while attempting to activate 'Elsa.Activities.Signaling.Services.Signaler'.)'
```

**Reply by sfmskywalker** [reply]
Crap. I think I might know. Will look into it first thing when I start today.

**Reply by sfmskywalker** [reply]
Should be good once the [2.0.0-rc2.96 build](https://ci.appveyor.com/project/sfmskywalker/elsa/builds/39191431) finishes.

**Reply by badihi** [reply]
Now I get this for both rc96 and rc98:

```
System.MissingMethodException: 'Method not found: 'System.Threading.Tasks.Task`1<System.Collections.Generic.IEnumerable`1<Elsa.Bookmarks.BookmarkFinderResult>> Elsa.BookmarkFinderExtensions.FindBookmarksAsync(Elsa.Bookmarks.IBookmarkFinder, System.String, System.Threading.CancellationToken)'.'
```

**Reply by sfmskywalker** [reply]
Can you push your changes (assuming it's part of https://github.com/badihi/elsa-http-async-mre) so I can see for myself? Might be package versioning thing.

**Comment by badihi** [comment]
Done

**Reply by sfmskywalker** [reply]
Perfect, thank you. The issue is that you're using different versions of Elsa packages: `2.0.0-rc2.96` and `2.0.0-rc2.73`. If you change all versions to be the same it will work. When I tried, I did get another runtime error which has to do with the SQLite database using an older schema. If you delete it, it should work.

**Reply by badihi** [reply]
Yes, my bad! It is now working beautifully as expected! Thanks!
